packages/server/src/users/database/UserSchema.ts

```diff
@@ -1,15 +1,80 @@
-import { Schema, SchemaFactory } from '@nestjs/mongoose';
+import { Prop, raw, Schema, SchemaFactory } from '@nestjs/mongoose';
 import { Exclude, Expose } from 'class-transformer';
 import { ObjectId } from 'mongodb';
-import { Document } from 'mongoose';
+import mongoose, { Document } from 'mongoose';
 
 type UserDocument = User & Document<ObjectId>;
 
+const USER_CONSTANTS = Object.freeze({
+  CARD: Object.freeze({
+    NUMBER: {
+      REGEX:
+        /^(?:4[0-9]{12}(?:[0-9]{3})?)|(?:3[47][0-9]{13})|(?:5[1-5][0-9]{14})|(?:6(?:011|5[0-9][0-9])[0-9]{12})|(?:(?:2131|1800|35\d{3})\d{11})|(?:3(?:0[0-5]|[68][0-9])[0-9]{11})$/,
+    },
+    CVC: { MIN_LENGTH: 3, MAX_LENGTH: 4 },
+  }),
+});
+
 @Exclude()
 @Schema({
   collection: 'users',
 })
 class User {
+  @Expose()
+  @Prop()
+  name: string;
+
+  @Expose()
+  @Prop()
+  surname: string;
+
+  @Expose()
+  @Prop({ required: true })
+  email: string;
+
+  @Expose()
+  @Prop({ required: true })
+  password: string;
+
+  @Expose()
+  @Prop({ required: true })
+  phoneNumber: string;
+
+  @Expose()
+  @Prop({ ref: 'Address' })
+  addressId: mongoose.Schema.Types.ObjectId;
+
+  @Expose()
+  @Prop(
+    raw({
+      number: { type: Number, required: true, match: USER_CONSTANTS.CARD.NUMBER.REGEX },
```

Comment:
Tutaj chyba będzie type "string", zgodnie z komentarzem który dał  moim pull requescie @tchojnacki .

https://github.com/CodersCamp2021-HK/CodersCamp2021.Project.Fullstack/pull/13#discussion_r815351013

Reply:
kurczę widziałem ten komentarz Tomka i wyleciało mi to z głowy później xD